projects/RabbitMQ.Client/client/impl/RabbitMQActivitySource.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using OpenTelemetry;
+using OpenTelemetry.Context.Propagation;
+
+using RabbitMQ.Client.Events;
+using RabbitMQ.Client.Framing.Impl;
+
+namespace RabbitMQ.Client
+{
+    internal class RabbitMQActivitySource
+    {
+        internal static ActivitySource source = new ActivitySource("RabbitMQ.Client", typeof(RabbitMQActivitySource).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion);
+        private static readonly TextMapPropagator Propagator = new CompositeTextMapPropagator(new TextMapPropagator[] { new TraceContextPropagator(), new BaggagePropagator() });
+
+
+        static RabbitMQActivitySource()
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+        }
+
+        internal static Activity Send(BasicPublish basicPublish, int bodySize)
+        {
+            Activity activity = StartRabbitMQActivity($"{basicPublish._routingKey} send", ActivityKind.Producer, Activity.Current?.Context ?? default);
+            if (activity != null && activity.IsAllDataRequested)
+            {
+                PopulateMessagingTags("send", basicPublish._routingKey, basicPublish._exchange, 0, null, bodySize, activity);
+            }
+
+            return activity;
+        }
+
+        internal static Activity Receive(string routingKey, string exchange, ulong deliveryTag, IReadOnlyBasicProperties readOnlyBasicProperties, int bodySize)
```

Comment:
I'd rename this to "HandleDelivery" because while consumers receive deliveries, the majority of the time spent will be spent in consumer delivery handler (that is, application-specific) code, not on the "receiving" code path in the client.

Reply:
It's just called `receive` since it is setting the Span name to `$"{routingKey} receive` since it's emitted when client receives a message, before it is handed over to a consumer. The consumer handler will do their processing within the `$"{routingKey} process` Span.

Reply:
Those names are the standardize OpenTelemetry Span names for messaging systems b.t.w: https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/messaging/

Reply:
I do see though after looking at the conventions that maybe the Exchange/Queue should be used in the span names and not the routing key. Does that make more sense @cijothomas?

Reply:
Ah, so this indeed is the delivery handler path in the client not including the consumer handler part. Thanks. Good to know that there are standard names for this.